datafusion/functions/src/math/tan.rs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Math function: `tan()`.
+
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow::datatypes::DataType;
+use arrow_array::{ArrayRef, Float32Array, Float64Array};
+use datafusion_common::{exec_err, plan_datafusion_err, DataFusionError, Result};
+use datafusion_expr::utils::generate_signature_error_msg;
+use datafusion_expr::Volatility;
+use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature};
+
+#[derive(Debug)]
+pub struct TanFunc {
+    signature: Signature,
+}
+
+impl TanFunc {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::uniform(
+                1,
+                vec![DataType::Float64, DataType::Float32],
+                Volatility::Immutable,
+            ),
```

Comment:
> Except for the `generate_signature_error_msg`, everything else looks good to me!

I'm thinking about if the signature has a fixed number of args to 1, is it safe to ignore length checking of `arg_types` in `ScalarUDFImpl::return_type`?

Reply:
I think it's safe, that's exactly what was done in [PR 9401](https://github.com/apache/arrow-datafusion/pull/9401/files#diff-2f2aa16b9b3f9d822cac85166bebf32e4f199e0ba5bd7a62f8583181a9487440). We can verify it through a test in sqllogictest.